python/tests/test_integration_schema.py

```diff
@@ -0,0 +1,2617 @@
+        NestedField(field_id=2, name="bar", field_type=IntegerType(), required=True),
+        NestedField(field_id=3, name="baz", field_type=BooleanType(), required=False),
+        schema_id=1,
+        identifier_field_ids=[2],
+    )
+
+
+@pytest.mark.integration
+def test_rename_simple_nested(catalog: Catalog) -> None:
+    tbl = _create_table_with_schema(
+        catalog,
+        Schema(
+            NestedField(
+                field_id=1,
+                name="foo",
+                field_type=StructType(NestedField(field_id=2, name="bar", field_type=StringType())),
+                required=True,
+            ),
+        ),
+    )
+
+    with tbl.update_schema() as schema_update:
+        schema_update.rename_column("foo.bar", "vo")
+
+    assert tbl.schema() == Schema(
+        NestedField(
+            field_id=1,
+            name="foo",
+            field_type=StructType(NestedField(field_id=2, name="vo", field_type=StringType())),
+            required=True,
+        ),
+    )
+
+
+@pytest.mark.integration
+def test_rename(catalog: Catalog) -> None:
+    tbl = _create_table_with_schema(
+        catalog,
+        Schema(
+            NestedField(
+                field_id=1,
+                name="location_lookup",
+                field_type=MapType(
+                    type="map",
+                    key_id=5,
+                    key_type=StringType(),
+                    value_id=6,
+                    value_type=StructType(
+                        NestedField(field_id=7, name="x", field_type=FloatType(), required=False),
+                        NestedField(field_id=8, name="y", field_type=FloatType(), required=False),
+                    ),
+                    value_required=True,
+                ),
+                required=True,
+            ),
+            NestedField(
+                field_id=2,
+                name="locations",
+                field_type=ListType(
+                    type="list",
+                    element_id=9,
+                    element_type=StructType(
+                        NestedField(field_id=10, name="x", field_type=FloatType(), required=False),
+                        NestedField(field_id=11, name="y", field_type=FloatType(), required=False),
+                    ),
+                    element_required=True,
+                ),
+                required=True,
+            ),
+            NestedField(
+                field_id=3,
+                name="person",
+                field_type=StructType(
+                    NestedField(field_id=12, name="name", field_type=StringType(), required=False),
+                    NestedField(field_id=13, name="leeftijd", field_type=IntegerType(), required=True),
+                ),
+                required=False,
+            ),
+            NestedField(field_id=4, name="foo", field_type=StringType(), required=True),
+            schema_id=0,
+            identifier_field_ids=[],
```

Comment:
Could add identifier fields to test nested identifier rename here.

Reply:
This was a great test case. With case-insensitive the identifier-field-ids were not properly updated